### PR TITLE
Feature/we 578 obtain director DOBs

### PR DESCRIPTION
**Pull request by duncannixon**
The purpose of this change is to add the Director Date of Birth screen to the application.

The complication with this change was that the Directors in the list are variable in number, therefore it was necessary to validate dynamically created text boxes. The validation didn't fit our existing framework It was necessary to perform the validation in the controller rather than in the Joi validators. It was also necessary to modify the validators so that they are now instantiated rather than being static, thereby allowing custom error messages to be created dynamically.

### CODE REVIEW

File: src/controllers/directorDateOfBirth.controller.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        }
+      }
+    } else {
+      errors = {
+        data: {
+          details: []
+        }
+      }
+
+      // Validate the entered DOB for each director
```
Comment (ajaxscape) [inline src/controllers/directorDateOfBirth.controller.js]: You could simplify by defining:

`const directorDobYear = `director-dob-day-${i}`
`

File: src/routes/confidentiality.route.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ const Constants = require('../constants')
 const {CONFIDENTIALITY, TASK_LIST} = Constants.Routes
 const ConfidentialityController = require('../controllers/confidentiality.controller')
 const ConfidentialityValidator = require('../validators/confidentiality.validator')
-const controller = new ConfidentialityController(CONFIDENTIALITY, true, TASK_LIST, ConfidentialityValidator)
+const controller = new ConfidentialityController(CONFIDENTIALITY, true, TASK_LIST, new ConfidentialityValidator())
```
Comment (ajaxscape) [inline src/routes/confidentiality.route.js]: Something to discuss and possibly refactor later, but I think how you have passed in an instance of the validator here is probably how we should be setting up every route.

File: src/views/directorDateOfBirth.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+        <p id="dob-explanation">
+          We use the date of birth to identify the person when we do security checks. We won’t include it in the public permit register.
+        </p>
+
+        <fieldset>
+          <legend id="dob-visually-hidden" class="visuallyhidden">Date of birth for director - we only need the date in the month</legend>
+
+          <dl id="dates-of-birth" class="dates-of-birth">
+            {{#if hasDirectors }}
```
Comment (ajaxscape) [inline src/views/directorDateOfBirth.html]: Possibly no need for hasDirectors:
{{#if directors.length}}

File: src/controllers/directorDateOfBirth.controller.js (hunk 1 above)
Comment (duncannixon) [reply src/controllers/directorDateOfBirth.controller.js]: Done (I called it directorDobField)

File: src/routes/confidentiality.route.js (hunk 2 above)
Comment (duncannixon) [reply src/routes/confidentiality.route.js]: To be refactored later.

**Pull request merged**